presto-router/src/test/java/com/facebook/presto/router/TestRouterAuthentication.java

```diff
@@ -0,0 +1,240 @@
+
+        PasswordAuthenticatorManager passwordAuthenticatorManager = injector.getInstance(PasswordAuthenticatorManager.class);
+        PasswordAuthenticatorFactory authFactory = new FileAuthenticatorFactory();
+        passwordAuthenticatorManager.addPasswordAuthenticatorFactory(authFactory);
+        passwordAuthenticatorManager.setRequired();
+        passwordAuthenticatorManager.loadPasswordAuthenticator();
+
+        lifeCycleManager = injector.getInstance(LifeCycleManager.class);
+        httpServerInfo = injector.getInstance(HttpServerInfo.class);
+        httpClient = new JettyHttpClient(
+                new HttpClientConfig()
+                        .setTrustStorePath(truststorePath.toAbsolutePath().toString())
+                        .setTrustStorePassword(jksPassword));
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDownServer()
+    {
+        if (authenticatorPropertiesFile != null) {
+            try {
+                authenticatorPropertiesFile.toFile().delete();
+            }
+            catch (Exception e) {
+                // ignore
+            }
+        }
+        if (lifeCycleManager != null) {
+            lifeCycleManager.stop();
+        }
+    }
+
+    @DataProvider(name = "routerEndpoints")
+    public Object[][] routerEndpoints()
+    {
+        return new Object[][] {
+                {"GET", "/"},
+                {"GET", "/v1/info"},
+                {"GET", "/v1/cluster"},
+                {"POST", "/v1/statement"},
+                // TODO: The UI is intentionally left out because the airlift incorrectly configures
+                //  some static resources
+                // {"GET", "/ui/"}
+        };
+    }
+
+    @Test(dataProvider = "routerEndpoints")
+    public void testEndpointWithoutAuthentication(String method, String endpoint)
+            throws Exception
+    {
+        Request request = Request.builder()
+                .setUri(UriBuilder.fromUri(httpServerInfo.getHttpsUri()).path(endpoint).build())
+                .setMethod(method)
+                .build();
+        httpClient.execute(request, new ResponseHandler<Void, Exception>()
+        {
+            @Override
+            public Void handleException(Request request, Exception exception)
+                    throws Exception
+            {
+                throw exception;
+            }
+
+            @Override
+            public Void handle(Request request, Response response)
+                    throws Exception
+            {
+                assertEquals(response.getStatusCode(), 401, format("request to %s should not have been successful. Expected 401, got: %d%n%s",
+                        request.getUri(), response.getStatusCode(), new String(ByteStreams.toByteArray(response.getInputStream()))));
+                return null;
+            }
+        });
+    }
+
+    @Test(dataProvider = "routerEndpoints")
+    public void testEndpointWithAuthentication(String method, String endpoint)
```

Comment:
Lets add another parameter here called `expectedAuthenticatdResponse`

For the `POST` to `/v1/statement` the value would be a `400`, for the others it would be `200`
We can then convert the -

```java
assertNotEquals(response.getStatusCode(), 401 ...
```

to 

```java
assertEquals(response.getStatusCode(), expectedAuthenticatdResponse ...
```

Reply:
I would have done this but I use the same data provider for the positive case, and I didn't want to have unused parameters in the data provider

Reply:
That's a no-op for the most part. TestNg will drop them if they are not referenced in the method definition, I think

Reply:
Fixed by adding a new value in the data provider for expected authenticated response code, and then adding the additional parameters in the methods. For the test without authentication I added `int unused`